lib/logzio-nodejs.js

```diff
@@ -1,20 +1,25 @@
 var request = require('request');
 var stringifySafe = require('json-stringify-safe');
 var _ = require('lodash');
+var dgram = require('dgram');
 
 exports.version = require('../package.json').version;
 
 var LogzioLogger = function (options) {
     if (!options || !options.token)
         throw new Error('You are required to supply a token for logging.');
 
-    if (options.protocol && options.protocol != 'http' && options.protocol != 'https')
-        throw new Error('Invalid protocol defined');
+    if (options.protocol && ['http', 'https', 'udp'].indexOf(options.protocol) === -1)
+        throw new Error('Invalid protocol defined. Valid options are http, https or udp');
+
+    if (options.protocol === 'udp') {
+        this.udpClient = dgram.createSocket('udp4');
+    }
 
     this.token = options.token;
     this.protocol = options.protocol || 'http';
     this.host = options.host || 'listener.logz.io';
-    this.port = this.protocol == 'http' ? 8070 : 8071;
+    this.port = this.protocol === 'udp' ? 5050 : (this.protocol === 'http' ? 8070 : 8071);
```

Comment:
I think it's nicer to have a map of protocols to ports. 
We can re-use this map to validate allowed protocols as well above